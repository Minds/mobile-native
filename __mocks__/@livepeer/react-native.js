// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
