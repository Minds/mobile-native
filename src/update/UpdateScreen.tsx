//@ts-nocheck
import React, { Component } from 'react';

import { Button, Text, Image, TextInput, StyleSheet, View } from 'react-native';

import { observer } from 'mobx-react';
import * as Progress from 'react-native-progress';

import { CommonStyle as CS } from '../styles/Common';
import updateService from '../common/services/update.service';
import colors from '../styles/Colors';
import { ComponentsStyle } from '../styles/Components';
import i18n from '../common/services/i18n.service';

@observer
export default class UpdatingScreen extends Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    return (
      <View style={[CS.backgroundWhite, CS.padding, CS.flexContainer]}>
        <Text style={[CS.fontXXL, CS.textCenter, CS.padding]}>
          {i18n.t('update.title', { version: updateService.version })}
        </Text>
        <View style={[CS.centered, CS.flexContainer]}>
          <View>
            <Image
              resizeMode={'contain'}
              style={ComponentsStyle.logo}
              source={require('../assets/logos/logo.png')}
            />
            <Progress.Bar
              progress={updateService.progress / 100}
              width={null}
              color={colors.primary}
            />
            <Text
              style={[
                CS.fontM,
                CS.fontMedium,
                CS.textCenter,
                CS.colorDark,
                CS.marginTop,
              ]}>
              {i18n.t('downloading')} {updateService.progress}%
            </Text>
          </View>
        </View>
      </View>
    );
  }
}
