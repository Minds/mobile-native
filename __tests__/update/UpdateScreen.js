import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import UpdateScreen from '../../src/update/UpdateScreen';

jest.mock('../../src/common/services/update.service');

describe('Update Screen Component', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<UpdateScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
