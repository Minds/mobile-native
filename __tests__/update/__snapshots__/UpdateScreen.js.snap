// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Screen Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#F6F7F7",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "justifyContent": "center",
        },
        Object {
          "padding": 20,
        },
        Object {
          "borderRadius": 14,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/logos/bulb.png",
        }
      }
      style={
        Object {
          "height": 90,
          "width": 80,
        }
      }
    />
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#43434D",
            },
            Object {
              "fontFamily": "Roboto",
            },
            Array [
              Object {
                "fontSize": 22,
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "paddingVertical": 12,
              },
            ],
          ]
        }
      >
        Updating to v
      </Text>
      <View
        animated={true}
        animationConfig={
          Object {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={false}
        indeterminateAnimationDuration={1000}
        onLayout={[Function]}
        progress={0}
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderColor": "#1B85D6",
              "borderRadius": 4,
              "borderWidth": 1,
              "overflow": "hidden",
              "width": null,
            },
            undefined,
          ]
        }
        useNativeDriver={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#1B85D6",
              "height": 6,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateX": 1,
                },
                Object {
                  "scaleX": 0.0001,
                },
              ],
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#43434D",
            },
            Object {
              "fontFamily": "Roboto",
            },
            Array [
              Object {
                "fontSize": 14,
              },
              Object {
                "fontFamily": "Roboto-Medium",
                "fontWeight": "500",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "#43434D",
              },
              Object {
                "marginTop": 8,
              },
            ],
          ]
        }
      >
        Downloading
         
        0
        %
      </Text>
    </View>
  </View>
</View>
`;
